src/aeso_ast_infer_types.erl

```diff
@@ -2910,6 +2912,12 @@ unify1(_Env, {uvar, _, R}, {uvar, _, R}, _Variance, _When) ->
     true;
 unify1(_Env, {uvar, _, _}, {fun_t, _, _, var_args, _}, _Variance, When) ->
     type_error({unify_varargs, When});
+unify1(_Env, {id, Ann, "hole"}, B, _Variance, _When) ->
```

Comment:
This doesn't seem right. Hole is an expression, not a type. Holes should have normal types as anything else and an error should be reported in infer_expr.